pinot-core/src/main/java/org/apache/pinot/core/indexsegment/mutable/MutableSegmentImpl.java

```diff
@@ -514,6 +518,32 @@ private void addForwardIndex(GenericRow row, int docId, Map<String, Object> dict
 
         numValuesInfo.updateMVEntry(dictIds.length);
       }
+
+      // Update min/max value for columns
+      BaseMutableDictionary dictionary = _dictionaryMap.get(column);
+      if (dictionary != null) {
+        _minValueMap.put(column, dictionary.getMinVal());
+        _maxValueMap.put(column, dictionary.getMaxVal());
+        continue;
+      }
+      if (value == null) {
```

Comment:
This check is redundant (1. value cannot be null; 2. already handled in the instanceof check)